azure/terraform/services/argo.tf

```diff
@@ -17,10 +28,54 @@ locals {
 resource "null_resource" "argo-quick-start-installation" {
   count = var.deploy_argo ? 1 : 0
   triggers = {
-    cmd = local._kubectl_cmd
+    cmd = local._argo_cmd
+  }
+  provisioner "local-exec" {
+    interpreter = local.is_windows ? ["PowerShell"] : null
+    command     = local.is_windows ? "$env:KUBECONFIG='${local_file.kubeconfig.filename}'; ${local._argo_cmd}" : "KUBECONFIG=${local_file.kubeconfig.filename} ${local._argo_cmd}"
+  }
+}
+
+resource "null_resource" "argo-events-quick-start" {
+  count = var.deploy_argo_events ? 1 : 0
+  triggers = {
+    cmd = local._argo_events_cmd
+  }
+  provisioner "local-exec" {
+    interpreter = local.is_windows ? ["PowerShell"] : null
+    command     = local.is_windows ? "$env:KUBECONFIG='${local_file.kubeconfig.filename}'; ${local._argo_events_cmd}" : "KUBECONFIG=${local_file.kubeconfig.filename} ${local._argo_events_cmd}"
+  }
+}
+
+resource "null_resource" "argo-events-service-accounts" {
```

Comment:
We use the null_resource + local_exec kubectl pattern for Argo Workflows originally because we want to just pull the quickstart YAML straight from an external source.

It looks like we are doing something similar here for Argo events installation "quickstart".

The additional stuff for `service_accounts.yaml` and `webhook_source.yaml` could in theory be done in native TF instead.  E.g. we bring up a full Metaflow stack on k8s cluster using native TF resources (kubernetes provider).  The main benefit would be standard TF resource state management (e.g. TF will know when things change, and act accordingly).

Though I understand if we don't want to spend the time transpiling it by hand.

If we do keep the null resource pattern, do we need to add some `depends_on` annotations?  I.e. the three local_execs for Argo events will race against each other - not sure if ordering matters.

Reply:
I was thinking about this earlier today. I think for now adding a few `depends_on` annotations should be enough. Then if we want to go whole hog on TF later we can.

Reply:
I'll update the PR in a bit with the dependency annotations.